core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -0,0 +1,469 @@
+        } else {
+            unset($this->additionalLinkAttributes[$attribute]);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Set the link href attribute. Set the value to null to use the auto
+     * generated default.
+     */
+    public function setLinkHref(?string $uri): self
+    {
+        $this->setLinkAttribute('href', $uri);
+
+        return $this;
+    }
+
+    /**
+     * Set a custom light box uri. By default or if the argument is set to
+     * null the uri will be automatically determined from the meta data
+     * or base resource.
+     *
+     * For this setting to take effect make sure you enabled the creation of a
+     * light box by calling `enableLightBox()`.
+     */
+    public function setLightBoxUri(?string $uri): self
+    {
+        $this->lightBoxUri = $uri;
+
+        return $this;
+    }
+
+    /**
+     * Set a size configuration that will be applied to the light box image (if
+     * available).
+     *
+     * For this setting to take effect make sure you enabled the creation of a
+     * light box by calling `enableLightBox()`.
+     *
+     * @param int|string|array|PictureConfiguration $size A picture size configuration or reference
+     */
+    public function setLightBoxSize($size): self
+    {
+        $this->lightBoxSizeConfiguration = $size;
+
+        return $this;
+    }
+
+    /**
+     * Set a custom light box group id. By default or if the argument is set to
+     * null an id will be generated.
+     *
+     * For this setting to take effect make sure you enabled the creation of a
+     * light box by calling `enableLightBox()`.
+     */
+    public function setLightBoxGroupIdentifier(?string $identifier): self
+    {
+        $this->lightBoxGroupIdentifier = $identifier;
+
+        return $this;
+    }
+
+    /**
+     * Enable/disable creation of a light box image (if possible) and/or
+     * outputting the respective link attributes. This setting is disabled by
+     * default.
+     */
+    public function enableLightBox(bool $enable = true): self
+    {
+        $this->enableLightBox = $enable;
+
+        return $this;
+    }
+
+    /**
+     * Create a result object with the current settings.
+     */
+    public function build(): Figure
+    {
+        if (null === $this->filePath) {
+            throw new \LogicException('You need to set a resource before building the result.');
+        }
+
+        // Freeze settings to allow reusing this builder object.
+        $settings = clone $this;
+
+        $imageResult = $this->locator
+            ->get('contao.image.studio')
+            ->createImage($settings->filePath, $this->sizeConfiguration)
+        ;
+
+        // We're defining some values via a Closure to make their evaluation lazy
+        return new Figure(
+            $imageResult,
+            function (Figure $figure) use ($settings): ?MetaData {
+                return $this->onDefineMetaData($settings);
+            },
```

Comment:
I think binding the closure to the cloned object and removing the `$settings` parameter might make this more clear:
```suggestion
            \Closure::bind(
                function (): ?MetaData {
                    return $this->onDefineMetaData();
                },
                $settings
            ),
```

Reply:
This is much nicer! :heart: